src/lib.rs

```diff
@@ -978,6 +978,14 @@ impl<T: Clone + Integer + Signed> Signed for Ratio<T> {
     }
 }
 
+#[cfg(feature = "std")]
+use std::string::String;
+
+#[cfg(feature = "std")]
+trait AsDecimal {
```

Comment:
If the new trait is not `pub`, then this whole thing is invisible to users.

But there's no reason for a new trait at all -- it can just be an inherent `pub fn`.

Reply:
Removed the trait and made a `pub fn`.